### PR TITLE
Add design and status docs

### PR DESCRIPTION
**Pull request by software-dov**
Includes state of the world, pagination, LRO, high-level generator, and high level generated surface docs.

These are intended to assist anyone attempting to resurrect C++ GAPIC at a future date by providing insight into design goals, philosophies, and intended directions.

NOTE: the docs are provided AS IS and are not intended to be polished, comprehensive user or developer documentation. If the code and the docs disagree on some detail, the code is correct.

### CODE REVIEW

**Comment by michaelbausor** [comment]
Is there any value in adding more of the docs listed in the C++ design overview document? Or is this selection curated for relevance already?

**Comment by software-dov** [comment]
I'm intending to curate for relevance. Docs for landed features or primitives are of less immediate utility of someone were to resurrect the project, and for a few of them (decoration and CallContext in particular) the code has diverged from the doc so much that the doc would need to be rewritten to be helpful and not misleading.

This is a topic on which I'm open to persuasion, but curating all the docs implies a greater scope and a stronger commitment to the archival process than just recording what we've already written.

**Comment by michaelbausor** [comment]
SGTM

**Comment by software-dov** [comment]
Open (doc) review bump.
@vam-google @michaelbausor

File: docs/STATUS.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
+# C++ GAPIC State of the World #
```
Comment (michaelbausor) [inline docs/STATUS.md]: Can we add:
- Links to the other design docs added in this PR
- Can we link to https://aip.dev/client-libraries/4210 as the spec the project is working to?

Comment (software-dov) [reply docs/STATUS.md]: Done

**Pull request merged**